src/python/dxpy/bindings/dxapplet.py

```diff
@@ -58,6 +58,17 @@ def _get_run_input_common_fields(executable_input, **kwargs):
         if kwargs.get('instance_type') is not None:
             run_input["systemRequirements"] = instance_type_to_sys_reqs(kwargs['instance_type'])
 
+        if kwargs.get('merged_cluster_spec'):
```

Comment:
This part mostly lg, but I do not like how we handle instance types and cluster specifications separately here. Consider merging the block at L58 with this block, using one of the following

```
# Option 1 -- use functions

def resolve_system_requirements_input(instance_type=None, cluster_specification=None):
    system_requirements = {}
    # Code
    return system_requirements

--//--

run_input["systemRequirements"] = resolve_system_requirements_input(instance_type=kwargs.get("instance_type"), cluster_specification=kwargs.get("merged_cluster_spec"))
```

or

```
# Option 2 -- use classes

class SystemRequirementsDict(object):
    def __init__(self, ...): ...
    @classmethod
    def from_instance_type(cls, instance_type=None):
        return cls(...)
    @classmethod
    def from_cluster_specification(cls, cluster_specification=None):
        return cls(...)

    def __add__(self, srd):
        """Add only, raise on collisions"""
        if not isinstance(srd, SystemRequirementsDict): raise
        # Code
        return SystemRequirementsDict(...)

    def add_override(self, srd)
        # Code
        return SRD(...)

    def as_dict(self): ...
--//--

run_input["systemRequirements"] = (SystemRequirementsDict.from_instance_type(...) + SRD.from_cluster_specification(...)).as_dict()
```

Might improve readability and unit-testability

Reply:
As I reviewed, I think I started to lean towards option 2, but let me know your thoughts